### PR TITLE
Implement mail chunking - 0.0.7-beta1

### PR DESCRIPTION
**Pull request by derpixler**
Send Notification mails in chnks

### CODE REVIEW

File: inform-about-content.php [hunk 1]
```diff
@@ -131,53 +135,69 @@ public static function get_object() {
...
 			if ( function_exists( 'spl_autoload_register' ) )
 				spl_autoload_register( array( __CLASS__, 'load_class' ) );
 			else
 				$this->load_class( NULL );
-			
+
 			// change the default behaviour from outside
-			self::$default_opt_in = apply_filters( 'iac_default_opt_in', FALSE );
-			
+			self::$default_opt_in = apply_filters( 'iac_default_opt_in', TRUE );
```
Comment (dnaber-de) [inline inform-about-content.php]: Why changing the default behavior here?

File: inform-about-content.php [hunk 2]
```diff
@@ -131,53 +135,69 @@ public static function get_object() {
...
 			#apply a hook to get the current settings
 			add_filter( 'iac_get_options', array( $this, 'get_options' ) );
-			
+
 			add_action( 'admin_init', array( $this, 'localize_plugin' ), 9 );
-			
+
+			#log settings
+			add_filter( 'logfile_path', function(){ return __DIR__ . '/log/'; } );
+			add_filter( 'logfile_name', function(){ return 'app.log'; } );
```
Comment (dnaber-de) [inline inform-about-content.php]: What are these filters for? Not WordPress defaults and not used anywhere else? Deprecated?

File: inform-about-content.php [hunk 3]
```diff
@@ -220,34 +240,44 @@ public function localize_plugin() {
...
-			);
-			$user_addresses = array();
+			}else{
+
+				$meta_key      = $context . '_subscription';
+				$meta_value    = '0';
+				$meta_compare  = '!=';
+				$include_empty = TRUE;
+
+				if ( self::$default_opt_in === FALSE &&  self::$default_opt_in) {
```
Comment (dnaber-de) [inline inform-about-content.php]: What is this?

File: inform-about-content.php [hunk 4]
```diff
@@ -496,11 +533,15 @@ public function inform_about_comment( $comment_id = FALSE, $comment_status = FAL
...
 
+
+
 		/**
 		 * builds the header and sends mail
 		 *
 		 * @since 0.0.5 (2012.09.03)
-		 * @param string $to
+		 *
+		 * @param int $object_id post or comment id
```
Comment (dnaber-de) [inline inform-about-content.php]: Docblock does not match method signature

File: inform-about-content.php [hunk 5]
```diff
@@ -525,6 +588,92 @@ public function send_mail( $to, $subject = '', $message = '', $headers = array()
...
+				$chunk_size = $this->options['static_options']['mail_to_chunking']['chunksize'];
+
+				foreach( $to as $email_address ){
+
+					if( $count > $chunk_size ){
+						$chunk++;
+						$count = 0;
+					}
+
+					$mail_to_chunks[ $chunk ][] = $email_address;
```
Comment (dnaber-de) [inline inform-about-content.php]: Don't re-declare variables, even more when they are method parameters. What does `$mail_to_chunk` says here?

File: inform-about-content.php [hunk 6]
```diff
@@ -525,6 +588,92 @@ public function send_mail( $to, $subject = '', $message = '', $headers = array()
...
+			$mail_to_chunks = array_values( $mail_to_chunks );
+
+			wp_schedule_single_event( time() + $this->options['static_options']['schedule_interval'], 'iac_schedule_send_chunks', array( $object_id, $object_type, $mail_to_chunks )  );
+
+			return $to;
+
+		}
+
+		/**
+		 * @param json|string $chunks
```
Comment (dnaber-de) [inline inform-about-content.php]: Docblock FTW

File: inform-about-content.php [hunk 7]
```diff
@@ -525,6 +588,92 @@ public function send_mail( $to, $subject = '', $message = '', $headers = array()
...
+		 */
+		public function schedule_send_next_group( $object_id, $object_type, $mail_to_chunks ){
+
+			$this->modulate_next_group( $object_id, $object_type, $mail_to_chunks );
+
+		}
+
+		/**
+		 * @param string $type means posttype post or comment
+		 * @param array $chunks
```
Comment (dnaber-de) [inline inform-about-content.php]: Docblock 
please

File: inform-about-content.php [hunk 8]
```diff
@@ -525,6 +588,92 @@ public function send_mail( $to, $subject = '', $message = '', $headers = array()
...
+
+				}elseif( $object_type == 'comment' ){
+
+					$this->inform_about_comment( $object_id );
+
+				}
+
+			}else{
+
+				//logg error
```
Comment (dnaber-de) [inline inform-about-content.php]: Log error, or don't log it?

File: inform-about-content.php [hunk 9]
```diff
@@ -525,6 +609,95 @@ public function send_mail( $to, $subject = '', $message = '', $headers = array()
...
+			return $to;
+
+		}
+
+		/**
+		 * Cron callback
+		 *
+		 * @wp-hook iac_schedule_send_chunks
+		 *
+		 * @param json|string $chunks
```
Comment (dnaber-de) [inline inform-about-content.php]: Dockblock, FTW

File: inform-about-content.php [hunk 10]
```diff
@@ -6,38 +6,39 @@
...
-		# since 0.0.6
-		add_filter( 'iac_post_message',    'strip_shortcodes' );
-		add_filter( 'iac_comment_message', 'strip_shortcodes' );
-	}
+	# since 0.0.6
+	add_filter( 'iac_post_message',    'strip_shortcodes' );
+	add_filter( 'iac_comment_message', 'strip_shortcodes' );
+
+	# since 0.0.7
+	#add_filter( 'iac_single_email_address', 'mask_address' );
```
Comment (dnaber-de) [inline inform-about-content.php]: Please remove if not needed any longer.

Comment (derpixler) [inline inform-about-content.php]: i think we need this for the coming test.

**Pull request merged**